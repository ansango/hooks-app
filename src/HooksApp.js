import { Container } from "./components/Container";

import * as modules from "./components/";

const components = Object.entries(modules);

export const HooksApp = () => {
  return (
    <div className="App">
      <Container>
        <h1>Samples Hooks</h1>
      </Container>
      {components.map((module, index) => {
        const name = module[0];
        const Component = module[1];
        return (
          <Container key={index} title={name}>
            <Component />
          </Container>
        );
      })}
    </div>
  );
};
