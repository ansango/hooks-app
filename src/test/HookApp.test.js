import { shallow } from "enzyme";
import { HooksApp } from "../HooksApp";

describe("Test in HooksApp", () => {
  test("should render", () => {
    const wrapper = shallow(<HooksApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
