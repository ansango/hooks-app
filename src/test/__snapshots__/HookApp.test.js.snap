// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in HooksApp should render 1`] = `
<div
  className="App"
>
  <Container>
    <h1>
      Hooks Samples
    </h1>
  </Container>
  <Container
    key="0"
    title="CounterApp"
  >
    <CounterApp />
  </Container>
  <Container
    key="1"
    title="CounterWithCustomHook"
  >
    <CounterWithCustomHook />
  </Container>
  <Container
    key="2"
    title="SimpleForm"
  >
    <SimpleForm />
  </Container>
  <Container
    key="3"
    title="FormWithCustomHook"
  >
    <FormWithCustomHook />
  </Container>
  <Container
    key="4"
    title="MultipleCustomHooks"
  >
    <MultipleCustomHooks />
  </Container>
  <Container
    key="5"
    title="FocusScreen"
  >
    <FocusScreen />
  </Container>
  <Container
    key="6"
    title="RealExampleRef"
  >
    <RealExampleRef />
  </Container>
  <Container
    key="7"
    title="LayoutEffect"
  >
    <LayoutEffect />
  </Container>
  <Container
    key="8"
    title="Memorize"
  >
    <Memorize />
  </Container>
  <Container
    key="9"
    title="MemoHook"
  >
    <MemoHook />
  </Container>
  <Container
    key="10"
    title="CallbackHook"
  >
    <CallbackHook />
  </Container>
  <Container
    key="11"
    title="MemoComponent"
  >
    <MemoComponent />
  </Container>
  <Container
    key="12"
    title="TodoApp"
  >
    <TodoApp />
  </Container>
  <Container
    key="13"
    title="MainApp"
  >
    <MainApp />
  </Container>
</div>
`;
